niryo-proxy/tools.js

```diff
@@ -17,10 +17,40 @@ class ArmConfig {
     }
 }
 
+class RobotState {
+    constructor(name, armStateMsg) {
+        this.name = name
+        this.x = armStateMsg.position.x
+        this.y = armStateMsg.position.y
+        this.z = armStateMsg.position.z
+        this.yaw = armStateMsg.rpy.yaw
+        this.pitch = armStateMsg.rpy.pitch
+        this.roll = armStateMsg.rpy.roll
+    }
+
+    equals(other) {
```

Comment:
Can be more generic:

```suggestion
equals(other) {
    // Create arrays of property names
    var aProps = Object.getOwnPropertyNames(a);
    var bProps = Object.getOwnPropertyNames(b);

    // If number of properties is different,
    // objects are not equivalent
    if (aProps.length != bProps.length) {
        return false;
    }

    for (var i = 0; i < aProps.length; i++) {
        var propName = aProps[i];

        // If values of same property are not equal,
        // objects are not equivalent
        if (a[propName] !== b[propName]) {
            return false;
        }
    }

    // If we made it this far, objects
    // are considered equivalent
    return true;
}
```